### PR TITLE
Add price quantity selection popup for product details screen

### PR DESCRIPTION
**Pull request by swapnilkamat23**
### Description -
This PR:
1. Adds price, quantity, tag, bottomsheet reusable(common) components
2. Adds price quantity selection popup for product details screen
3. Organizes exports for common components and screen better using index.ts files

### Screenshots -
IMG_8611

IMG_8614

### CODE REVIEW

**Review by vaibhavnagda — COMMENTED** [review]
Looks good. 👌  I got some import errors because of there is a uppercase `P` in Product directory whereas the imports are using lower case `p`. Same for Category imports too.

File: app/views/components/common/Quantity.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { StyleSheet, Text } from "react-native";
+
+import colors from "../../styles/colors";
+
+export enum QUANTITY_UNITS {
+  KG = "kg",
+  GM = "gm",
+  ML = "ml",
+  L = "l",
+}
```
Comment (vaibhavnagda) [inline app/views/components/common/Quantity.tsx]: Should this be added into constants/enums?

File: app/views/components/common/Tag.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { StyleSheet, Text, View } from "react-native";
+
+import colors from "../../styles/colors";
+
+export enum TAG_TYPE {
+  CUSTOM = "custom",
+  BEST_VALUE = "Best value",
+  POPULAR = "Popular",
+}
```
Comment (vaibhavnagda) [inline app/views/components/common/Tag.tsx]: Same as above?

File: app/views/components/common/Quantity.tsx (hunk 1 above)
Comment (swapnilkamat23) [reply app/views/components/common/Quantity.tsx]: Yes, will move it over

File: app/views/components/common/Tag.tsx (hunk 2 above)
Comment (swapnilkamat23) [reply app/views/components/common/Tag.tsx]: Yes, will move it over

**Comment by swapnilkamat23** [comment]
> Looks good. 👌 I got some import errors because of there is a uppercase `P` in Product directory whereas the imports are using lower case `p`. Same for Category imports too.
 

Yeah, I did a fresh pull and ran into the same issue. Not sure why git didn't push changes for directory renaming. Will try pushing those changes again. Thanks for calling that out.

File: app/views/components/common/Quantity.tsx (hunk 1 above)
Comment (swapnilkamat23) [reply app/views/components/common/Quantity.tsx]: Updated

File: app/views/components/common/Tag.tsx (hunk 2 above)
Comment (swapnilkamat23) [reply app/views/components/common/Tag.tsx]: Updated

**Review by vaibhavnagda — APPROVED** [review]

File: app/views/components/product/ProductDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+      quantity: 2,
+      quantityUnit: QUANTITY_UNITS.KG,
+    },
+    {
+      newPrice: 1050,
+      oldPrice: 1400,
+      quantity: 4,
+      quantityUnit: QUANTITY_UNITS.KG,
+      tagType: TAG_TYPE.CUSTOM,
+      tagContent: "25% off!",
```
Comment (ndedhia1210) [inline app/views/components/product/ProductDetails.tsx:54]: Who sets these two fields? Should they be on Backend too? 
Btw, what is Tag type?

**Review by ndedhia1210 — APPROVED** [review]

**Pull request merged**